### ISSUE TITLE
Ask about the code of SparseMoeBlock.moe_infer()

### ISSUE THREAD

**Issue by ChangeFWorld** [feizc/DiT-MoE#3]
Awesome work! I'm trying to train and i'm curious about why you implement a moe_infer() method? It seems that the forward() method in training stage has the same function.  

```python
if self.training:
            hidden_states = hidden_states.repeat_interleave(self.num_experts_per_tok, dim=0)
            y = torch.empty_like(hidden_states, dtype=hidden_states.dtype)
            for i, expert in enumerate(self.experts): 
                y[flat_topk_idx == i] = expert(hidden_states[flat_topk_idx == i]).float()
            y = (y.view(*topk_weight.shape, -1) * topk_weight.unsqueeze(-1)).sum(dim=1)
            y =  y.view(*orig_shape)
            y = AddAuxiliaryLoss.apply(y, aux_loss)
else:
            y = self.moe_infer(hidden_states, flat_topk_idx, topk_weight.view(-1, 1)).view(*orig_shape)

**Comment by feizc** [comment]
Hi, actually, the implementation here is based on Deepseek-MoE in this [line](https://huggingface.co/deepseek-ai/deepseek-moe-16b-base/blob/main/modeling_deepseek.py#L381).  It avoids redundant computation, such as auxiliary loss.

**Comment by ChangeFWorld** [comment]
Thank you for reply!
